src/System.Device.Gpio/System/Device/Board.cs

```diff
@@ -0,0 +1,246 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Device.Analog;
+using System.Device.Boards;
+using System.Device.Gpio;
+using System.Device.I2c;
+using System.Device.Pwm;
+using System.Device.Spi;
+using System.IO;
+using System.Text;
+
+namespace System.Device
+{
+    public abstract class Board : MarshalByRefObject, IDisposable
+    {
+        private readonly PinNumberingScheme _defaultNumberingScheme;
+        private readonly object _pinReservationsLock;
+        private readonly Dictionary<int, PinReservation> _pinReservations;
+        private bool _initialized;
+        private bool _disposed;
+
+        protected Board(PinNumberingScheme defaultNumberingScheme)
+        {
+            _defaultNumberingScheme = defaultNumberingScheme;
+            _pinReservations = new Dictionary<int, PinReservation>();
+            _pinReservationsLock = new object();
+            _initialized = false;
+            _disposed = false;
+        }
+
+        ~Board()
+        {
+            Dispose(false);
```

Comment:
Why do we need a finalizer?

Reply:
I added it for completeness, and I think it makes sense here: All of the derived (and aggregated) classes implement IDisposable and potentially allocate unmanaged resources such as device driver handles. These should be cleaned up even if the user forgets to call Dispose.

Reply:
In that case the derived and aggregated classes are the ones that should have the finalizer (if applicable), not this one. In general in .NET we actually try to not use finalizers at all when possible, and instead use the Dispose pattern and always using SafeHandles whenever we have a type that holds native resources.

Reply:
you may find this page helpful with guidance about Dispose patterns and finalizers: https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose